test/avrora/storage/registry_test.exs

```diff
@@ -161,6 +161,84 @@ defmodule Avrora.Storage.RegistryTest do
 
       Application.put_env(:avrora, :registry_url, registry_url)
     end
+
+    test "when registry auth is malformed" do
+      registry_auth = Application.get_env(:avrora, :registry_auth)
+      Application.put_env(:avrora, :registry_auth, {:basic, "user", "pass", "unknown"})
+
+      assert Registry.put("anything", ~s({"type":"string"})) == {:error, :malformed_registry_auth}
+
+      Application.put_env(:avrora, :registry_auth, registry_auth)
+    end
+
+    test "when registry auth file not exists" do
+      registry_auth = Application.get_env(:avrora, :registry_auth)
+      Application.put_env(:avrora, :registry_auth, {:basic, "unknown"})
+
+      output =
+        capture_log(fn ->
+          assert Registry.put("anything", ~s({"type":"string"})) ==
+                   {:error, :no_such_registry_auth_file}
+        end)
+
+      output =~ "no such registry auth file found unknown"
+
+      Application.put_env(:avrora, :registry_auth, registry_auth)
+    end
+
+    test "when registry auth basic file is configured" do
+      Avrora.HTTPClientMock
+      |> expect(:post, fn url, payload, headers: headers, content_type: _ ->
+        base64 = :base64.encode_to_string("avrora_username:avrora_password")
+
+        assert url == "https://reg.loc/subjects/io.confluent.Payment/versions"
+        assert payload == json_schema()
+        assert headers == [{'Authorization', 'Basic #{base64}'}]
+
+        {:ok, %{"id" => 1}}
+      end)
+
+      registry_url = Application.get_env(:avrora, :registry_url)
+      registry_auth = Application.get_env(:avrora, :registry_auth)
+      Application.put_env(:avrora, :registry_url, "https://reg.loc")
+      Application.put_env(:avrora, :registry_auth, {:basic, "./test/fixtures/auth"})
+
+      {:ok, schema} = Registry.put("io.confluent.Payment", json_schema())
+
+      assert schema.id == 1
+      assert is_nil(schema.version)
+      assert schema.full_name == "io.confluent.Payment"
+
+      Application.put_env(:avrora, :registry_url, registry_url)
+      Application.put_env(:avrora, :registry_auth, registry_auth)
+    end
+
+    test "when registry auth basic is configured" do
+      Avrora.HTTPClientMock
+      |> expect(:post, fn url, payload, headers: headers, content_type: _ ->
+        base64 = :base64.encode_to_string("avrora_username:avrora_password")
```

Comment:
Please replace dynamic computations with exact values. In this case, `base64` value can be computed and just injected at line 223 as complete value